### PR TITLE
Speed up PolylineBatch tests

### PR DESCRIPTION
**Pull request by OmarShehata**
Do not merge yet, opening for feedback. 

@likangning93 this does some cleanup to `DataSources/StaticGroundPolylinePerMaterialBatch`. Initially the tests were timing out, and failing subsequent tests because resources were only getting cleaned up if the test succeeded. 

1. I added an `afterEach` to ensure any added primitives are removed.
2. I made the geometry all the tests use much smaller, with a much bigger granularity.
3. I made all the primitives synchronous. 

For 3, I had to tweak the API to accept an option to set the created primitive as synchronous. I noticed that `StaticGroundPolylinePerMaterialBatch.prototype.update` calls `Batch.prototype.update` so I added the default value to the former. This should work fine unless there are any instances when `Batch.prototype.update` is called directly and not through the former.

I don't have before/after timing results because the before takes so long that the tests all timeout on this machine, this should be a significant boost.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @OmarShehata!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/speed-batch-tests/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by OmarShehata** [comment]
To clarify, I confirmed that these tests actually fail on computers that are not high-end, and the only reason they pass on Travis I believe is because of:

```
if (!GroundPolylinePrimitive.isSupported(scene)) {
            // Don't fail if GroundPolylinePrimitive is not supported
            return;
        }
```

File: Source/DataSources/StaticGroundPolylinePerMaterialBatch.js [hunk 1]
```diff
@@ -133,7 +135,7 @@ define([
 
                 primitive = new GroundPolylinePrimitive({
                     show : false,
-                    asynchronous : true,
+                    asynchronous : asynchronous,
```
[the remark this answers is not in the document]
Comment (OmarShehata) [reply Source/DataSources/StaticGroundPolylinePerMaterialBatch.js]: That's just the case where it retries on the next frame right? I don't see a super obvious way to do it and I'm not sure it's worth adding a separate spec for right now.

**Comment by OmarShehata** [comment]
@likangning93 I added your feedback, and all tests are passing locally.

File: Source/DataSources/StaticGroundPolylinePerMaterialBatch.js [hunk 2]
```diff
@@ -334,7 +339,7 @@ define([
 
         var isUpdated = true;
         for (i = 0; i < items.length; i++) {
-            isUpdated = items[i].update(time) && isUpdated;
+            isUpdated = items[i].update(time, this._asynchronous) && isUpdated;
```
Comment (likangning93) [inline Source/DataSources/StaticGroundPolylinePerMaterialBatch.js]: Should this still be here?

File: Source/DataSources/StaticGroundPolylinePerMaterialBatch.js [hunk 3]
```diff
@@ -59,6 +61,8 @@ define([
         this.subscriptions = new AssociativeArray();
         this.showsUpdated = new AssociativeArray();
         this.zIndex = zIndex;
+
+        this._asynchronous = defaultValue(asynchronous, true);
```
Comment (likangning93) [inline Source/DataSources/StaticGroundPolylinePerMaterialBatch.js]: Small thing, but it doesn't look like we would expect this input to ever be undefined. Should be safe to just set it directly.

**Comment by likangning93** [comment]
@OmarShehata bump on a few more things

**Comment by OmarShehata** [comment]
Yes, thanks for the reminder @likangning93 ! It's been on my list. Will get to it later today.

**Comment by OmarShehata** [comment]
@likangning93 I took care of the cleanup issues. I feel like having one test be async would be a big waste, especially if each suite started having at least one async. I feel like that code path could be tested in tests that specifically have something to test with async, but otherwise we should stick with synchronous.

**Comment by OmarShehata** [comment]
@likangning93 this is ready to merge!

**Comment by likangning93** [comment]
Confirmed that these tests finish now instead of timing out on my n2920-based intel machine, yay!

**Pull request merged**